source/validate.cpp

```diff
@@ -332,14 +270,28 @@ void DebugInstructionPass(ValidationState_t& _,
   }
 }
 
+// Collects use-def info about an instruction's IDs.
+void ProcessIds(ValidationState_t& _, const spv_parsed_instruction_t& inst) {
+  if (inst.result_id) {
+    _.usedefs().AddDef(
+        {inst.result_id, inst.opcode,
+         std::vector<uint32_t>(inst.words, inst.words + inst.num_words)});
+  }
+  for (auto op = inst.operands; op != inst.operands + inst.num_operands; ++op) {
+    if (spvIsIdType(op->type))
+      _.usedefs().AddUse(inst.words[op->offset]);
```

Comment:
The multiple-ID-definitions check is [already covered](https://github.com/KhronosGroup/SPIRV-Tools/blob/master/source/binary.cpp#L478) by the underlying parser, so it's redundant anywhere in the validator.